### PR TITLE
Add a access(2) wrapper

### PR DESCRIPTION
**Pull request by snsmac**

### CODE REVIEW

File: src/unistd.rs [hunk 1]
```diff
@@ -2334,3 +2334,29 @@ mod setres {
...
+        /// Test for write permission.
+        W_OK;
+        /// Test for execute (search) permission.
+        X_OK;
+    }
+}
+
+/// Checks the file named by `path` for accessibility according to the flags given by `amode`
+/// See [access(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/access.html)
+#[inline]
```
Comment (asomers) [inline src/unistd.rs]: No need for `#[inline]`.  It's better to allow the compiler to decide.

File: test/test_unistd.rs [hunk 2]
```diff
@@ -574,3 +574,22 @@ fn test_symlinkat() {
...
+               nix::errno::Errno::ENOENT);
+}
+
+#[test]
+fn test_access_file_exists() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let path  = tempdir.path().join("does_exist.txt");
+    let _file = File::create(path.clone()).unwrap();
+    assert_eq!(
+        access(&path, AccessFlags::F_OK | AccessFlags::R_OK | AccessFlags::W_OK).unwrap(),
```
Comment (asomers) [inline test/test_unistd.rs]: Instead of unwrapping, just should just `assert!(access(...).is_ok())`

File: src/unistd.rs (hunk 1 above)
Comment (snsmac) [reply src/unistd.rs]: Ok, i did this because many other functions in unistd.rs also use inline

Comment (asomers) [reply src/unistd.rs]: Yeah, it's an anachronism that we haven't fully purged.

**Comment by snsmac** [comment]
@asomers I made the changes you requested

**Pull request merged**